repos/system_upgrade/common/actors/systemfacts/libraries/systemfacts.py

```diff
@@ -254,4 +259,51 @@ def _get_firewall_status(service_name):
 
 def get_firmware():
     firmware = 'efi' if os.path.isdir('/sys/firmware/efi') else 'bios'
+    if architecture.matches_architecture(architecture.ARCH_PPC64LE):
+        ppc64le_opal = bool(os.path.isdir('/sys/firmware/opal/'))
+        return FirmwareFacts(firmware=firmware, ppc64le_opal=ppc64le_opal)
     return FirmwareFacts(firmware=firmware)
+
+
+@aslist
+def _default_grub_info():
+    default_grb_fpath = '/etc/default/grub'
+    if not os.path.isfile(default_grb_fpath):
+        reporting.create_report([
+            reporting.Title('File "{}" does not exist!'.format(default_grb_fpath)),
+            reporting.Summary(
+                'Leapp detected "{}" does not exist. The file is essential for the in-place upgrade '
```

Comment:
Hmmm isn't it missing `.format(default_grb_path)` ?

Reply:
nice catch